libraries/AP_Logger/LogFile.cpp

```diff
@@ -1413,14 +1413,10 @@ void AP_Logger::Write_Current_instance(const uint64_t time_us,
             time_us             : time_us,
             voltage             : battery.voltage(battery_instance)
         };
-        for (uint8_t i = 0; i < ARRAY_SIZE(cells.cells); i++) {
+        for (uint8_t i = 0; i < MAVLINK_MSG_BATTERY_STATUS_FIELD_VOLTAGES_LEN; i++) {
             cell_pkt.cell_voltages[i] = cells.cells[i] + 1;
         }
         WriteBlock(&cell_pkt, sizeof(cell_pkt));
-
-        // check battery structure can hold all cells
-        static_assert(ARRAY_SIZE(cells.cells) == (sizeof(cell_pkt.cell_voltages) / sizeof(cell_pkt.cell_voltages[0])),
```

Comment:
This is the key to updating the struct info. If you don't do this additional cells may not be logged silently. The assert must continue to live.

Reply:
Humm, this will be more tricky then, but I will find a soltution.